rust-spec/lightclient/detection/detection.md

```diff
@@ -1,50 +1,68 @@
-*hd* (after performing skipping or sequential verification), it can
-use the  detector to probe the system for conflicting headers and
-increase the trust in *hd*. Instead of communicating with a single
-full node, communicating with several full nodes shall increase the
-likelihood to be aware of a fork (see [[accountability]] for
-discussion about forks) in case there is one.
-
-2) to support fork accountability: In the case when more than 1/3 of
-the voting power is held by faulty validators, faulty nodes may
-generate two conflicting headers for the same height. The goal of the
-detector is to learn about the conflicting headers by probing
-different full nodes. Once a detector has two conflicting headers,
-these headers are evidence of misbehavior. A natural extension is to
-use the detector within a monitor process (on a full node) that calls
-the detector on a sample (or all) headers (in parallel). (If the
-sample is chosen at random, this adds a level of probabilistic
-reasoning.) If conflicting headers are found, they are evidence that
-can be used for punishing processes.
-
-In this document we will focus onn strengthening the light client, and
-leave other uses of the detection mechanism (e.g., when run on a full
-node) to the future.
+# ***This an unfinished draft. Comments are welcome!***
+
+**TODO:** We will need to do small adaptations to the verification
+spec to reflect the semantics in the LightStore (verified, trusted,
+untrusted, etc. not needed anymore). In more detail:
+
+- The state of the Lightstore needs to go. Functions like `LatestVerified` can
+keep the name but will ignore state as it will not exist anymore.
+
+- verification spec should be adapted to the second parameter of
+`VerifyToTarget` 
+being a lightblock; new version number of function tag;
+
+- We should clarify what is the expectation of VerifyToTarget
+so if it returns TimeoutError it can be assumed faulty. I guess that
+VerifyToTarget with correct full node should never terminate with
+TimeoutError.
+
+- We need to introduce a new version number for the new
+specification. So we should decide how
+  to handle that.
+
+# Light Client Attack Detector
+
+In this specification, we strengthen the light client to be resistant
+against so-called light client attacks. In a light client attack, all
+the correct Tendermint full nodes agree on the sequence of generated
+blocks (no fork), but a set of faulty full nodes attack a light client
+by generating (signing) a block that deviates from the block of the
+same height on the blockchain. In order to do so, some of these faulty
+full nodes must have been validators before and violate
+[[TMBC-FM-2THIRDS]](TMBC-FM-2THIRDS-link), as otherwise, if
+[[TMBC-FM-2THIRDS]](TMBC-FM-2THIRDS-link) would hold,
+[verification](verification) would satisfy
+[[LCV-SEQ-SAFE.1]](LCV-SEQ-SAFE-link).
+
+An attack detector (or detector for short) is a mechanism that is used
+by the light client [supervisor](supervisor) after
```

Comment:
Supervisor link is broken.

Reply:
We don't have the files on master yet. I will fix that then.